docs/source/developer_guide/dependencies.rst

```diff
@@ -93,3 +93,19 @@ the following files need to be updated:
 
 *  `pyproject.toml <https://github.com/sktime/sktime/blob/main/pyproject.toml>`__,
    adding the dependency or version bounds in the ``dev`` dependency set.
+
+Maintenance of bounds for soft dependencies
+-------------------------------------------
+
+*  As of this writing, it has been decided that all soft-dependencies will have lower
+   and upper bounds specified mandatorily.
+*  The soft-dependencies will be specified in separate extras per each component of
+   ``sktime``, for example ``forecasting``, ``classification``, ``regression``, etc.
+*  It is possible to have different upper and lower bounds for a single package when
```

Comment:
is that advised? How is this handled in `all_extras`?

Reply:
In `all_extras`, a soft-dependency can only be mentioned once, so have to compatible with all other soft dependencies for all components. And that means even if one component, say `C1` can use package `P` from `L1` to `H1`, it'll be restricted to use `L2` (where `L2` >= `L1`) to `H2` (where `H2` <= `H1`) if component `C2` is incompatible with `P<L2` and/or `P>H2`.

My main reason to suggest per component extras was to avoid these type of restrictions. `C1` and `C2` then becomes independent, and we can test more freely about their own dependencies without worrying much (we still should not do very conflicting stuff, e.g. `forecasting` with `pandas<2` and `transformations` with `pandas>2` - `pandas` just for example) about compatibility with the other.

Reply:
makes sense